Changes for 1.3.0: python3 + fixes

Wescoeur opened the pull request.

Wescoeur committed:
Tests are now compatible with python3

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>

Wescoeur committed:
Remove empty dependency line

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>

ydirson commented on tests/conftest.py:
```diff
@@ -181,7 +189,7 @@ class Device(object):
 
     def __init__(self, buffer, fd):
         self._buffer = buffer
-        self._read_buffer = generate_aligned_random_buffer(len(buffer) / 1024, SECTOR_SIZE)
+        self._read_buffer = generate_aligned_random_buffer(int(len(buffer) / 1024), SECTOR_SIZE)
```
you likely want `//` instead, with the relevant `from __future__` for py2 compat

Wescoeur replied:
Right! Nice catch. :)

Wescoeur committed:
O_SYNC is no longer required in our tests

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>

Wescoeur committed:
Fix assert triggered in curl_cb_write

Normally when `curl_cb_write` is called we have a buffer
available to write the result of a GET request.
This callback is executed when nbdkit asks for a data chunk
to read on the HTTP server.

But there is an issue here... The `curl_cb_write` is also
executed after a PUT message on the server... It's the response of our write,
just a small message: "Ok.". And in this case, it was assumed that only
`curl_cb_read` was executed to send our chunk, but not `curl_cb_write` then.
We can therefore find ourselves in this function without having a buffer.

Luckily, we always try to execute a GET when the NBD server starts,
so a read buffer is assigned to our handle. Subsequently if a write request
is executed, the response from the server can be written in this buffer:
- But it's not really a chance, this buffer is managed by nbdkit,
  there is no guarantee that it is valid.
- And if it is valid, don't we write "Ok!" in the device? Which would be a problem...

So I managed to trigger this assertion and infer these issues by testing
our servers in a certain way: A small delay when starting the HTTP server that
caused read requests to the NBD server to fail. So no read buffer was attached to our handle.
And just after that, it became possible to execute a write request which triggered
the assertion.

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>

benjamreis commented on tests/conftest.py:
```diff
@@ -319,7 +327,7 @@ def clean():
         (nbd_server, nbd_path) = start_nbd_server('disk-test-mirror', RANDOM_BUFFER_SIZE_KIB * 1024)
-        fd = os.open(nbd_path, os.O_RDWR | os.O_DIRECT | os.O_SYNC)
+        fd = os.open(nbd_path, os.O_RDWR | os.O_DIRECT)
     except Exception:
```
The `O_SYNC` changes are not for Python 3 support.
